### PR TITLE
rename `getDataSpace` to `capacityND`

### PR DESCRIPTION
**Pull request by psychocoderHPC**
- Buffers:
  - rename `getDataSpace` to `capacityND`
- GridLayout
  - rename `getDataSpcae()` to `sizeND()`
  - rename `getGuard()` to `guardSizeND()`
  - rename `getDataSPaceWithoutGarding()` to `sizeWithoutGuardND()`

**Note:** I only renamed the functions, I did not update the variable names inside of function where the changed functions are used.

### CODE REVIEW

**Review by ikbuibui — DISMISSED** [review]

**Review by chillenzer — APPROVED** [review]
Okay. Very straightforward. Still one question: I'm totally with you that a "capacity" is a property of a buffer. Is it also a property of a grid layout? (Not sure if this is understandable via text, we can discuss offline if not.)

**Comment by psychocoderHPC** [comment]
> Okay. Very straightforward. Still one question: I'm totally with you that a "capacity" is a property of a buffer. Is it also a property of a grid layout? (Not sure if this is understandable via text, we can discuss offline if not.)

NO should be size for the grid layout. Search and replace bug. I will fix it.

**Comment by psychocoderHPC** [comment]
I updated the PR

File: include/pmacc/fields/operations/AddExchangeToBorder.hpp [hunk 1]
```diff
@@ -167,9 +167,9 @@ namespace pmacc
                      * @warning pmacc restriction: all dimension must have the some number of guarding
                      * supercells
                      */
-                    auto const numGuardSuperCells = destBuffer.getGridLayout().getGuard() / SuperCellSize::toRT();
+                    auto const numGuardSuperCells = destBuffer.getGridLayout().sizeND() / SuperCellSize::toRT();
```
Comment (ikbuibui) [inline include/pmacc/fields/operations/AddExchangeToBorder.hpp]:
```suggestion
                    auto const numGuardSuperCells = destBuffer.getGridLayout().guardSizeND() / SuperCellSize::toRT();
```

**Review by ikbuibui — CHANGES_REQUESTED** [review]

File: share/pmacc/examples/heatEquation2D/main.cpp [hunk 2]
```diff
@@ -107,7 +107,7 @@ auto main(int argc, char** argv) -> int
      *  takes in the grid layout - CELLS dataspace + guard, and num of SUPERCELLS in guard
      */
     pmacc::DataSpace<DIM2> guardingSuperCells{1, 1};
-    auto mapping = std::make_unique<MappingDesc>(layout.getDataSpace(), guardingSuperCells);
+    auto mapping = std::make_unique<MappingDesc>(layout.capacityND(), guardingSuperCells);
```
Comment (ikbuibui) [inline share/pmacc/examples/heatEquation2D/main.cpp]:
```suggestion
    auto mapping = std::make_unique<MappingDesc>(layout.sizeND(), guardingSuperCells);
```

File: share/pmacc/examples/gameOfLife2D/include/Simulation.hpp [hunk 3]
```diff
@@ -145,7 +145,7 @@ namespace gol
              * MappingDesc stores the layout regarding Core, Border and Guard     *
              * in units of SuperCells.                                            *
              * This is saved by init to be used by the kernel to identify itself. */
-            evo.init(layout.getDataSpace(), Space::create(1));
+            evo.init(layout.capacityND(), Space::create(1));
```
Comment (ikbuibui) [inline share/pmacc/examples/gameOfLife2D/include/Simulation.hpp]:
```suggestion
            evo.init(layout.sizeND(), Space::create(1));
```

**Review by ikbuibui — CHANGES_REQUESTED** [review]

File: share/pmacc/examples/heatEquation2D/main.cpp (hunk 2 above)
Comment (psychocoderHPC) [reply share/pmacc/examples/heatEquation2D/main.cpp]: done

**Review by ikbuibui — APPROVED** [review]

**Pull request merged**